src/pykeen/nn/emb.py

```diff
@@ -1068,3 +1070,103 @@ def forward(
         if indices is not None:
             x = x[indices]
         return x
+
+
+class NodePieceRepresentation(RepresentationModule):
+    """
+    Basic implementation of node piece decomposition [galkin2021]_.
+
+    .. note ::
+        This implementation currently only supports representation of entities by bag-of-relations.
+    """
+
+    #: the token representations
+    tokens: torch.FloatTensor
+
+    #: the entity-to-token mapping
+    assignment: torch.LongTensor
+
+    def __init__(
+        self,
+        *,
+        triples_factory: CoreTriplesFactory,
+        token_representation: EmbeddingSpecification,
+        aggregation: Callable[[torch.Tensor, int], torch.Tensor] = None,
+        k: int = 1,
+        **kwargs,
+    ):
+        """
+        Initialize the representation.
+
+        :param triples_factory:
+            the triples factory
+        :param token_representation:
+            the token representation specification
+        :param aggregation:
+            aggregation of multiple token representations to a single entity representation
```

Comment:
besides `torch.mean`, what other things might go here? Is `torch.sum` reasonable? Or anything else? Please add to docs :)

Reply:
The simple ones would be sum/mean/max.

We can also have trainable aggregations, e.g., `MLP(mean(MLP(tokens)))` (cf. [DeepSets](https://arxiv.org/abs/1703.06114)).

In theory, we could also have aggregations which result in differently shapes output, e.g. a concatenation of all token embeddings resulting in shape `(k * d,)`, but for this, we would need to adjust the code.

I tried to keep it simple for now :slightly_smiling_face:

Reply:
For reference: the paper uses either
 1) MLP that takes as input a concatenated hash vector R1×(k+m)d projecting it down to Rd;  or
2) Transformer encoder [42] with average pooling that takes as input an original sequence R(k+m)×d